src/util/strencodings.h

```diff
@@ -94,6 +96,24 @@ constexpr inline bool IsSpace(char c) noexcept {
     return c == ' ' || c == '\f' || c == '\n' || c == '\r' || c == '\t' || c == '\v';
 }
 
+/**
+ * Convert string to integral type T.
+ *
+ * @returns std::nullopt if the entire string could not be parsed, or if the
+ *   parsed value is not in the range representable by the type T.
+ */
+template <typename T>
+std::optional<T> ToIntegral(const std::string& str)
+{
+    static_assert(std::is_integral<T>::value);
+    T result;
+    const auto [first_nonmatching, error_condition] = std::from_chars(str.data(), str.data() + str.size(), result);
+    if (first_nonmatching != str.data() + str.size() || error_condition != std::errc{}) {
+        return std::nullopt;
+    }
+    return {result};
```

Comment:
Can you explain what the `{` are supposed to do? I think they can be removed, as they are confusing as-is. If you want them to check (for whatever reason) that result fits in type `T`, you will have to type `return T{result};`

Reply:
In this specific case `return result;` should be equivalent to `return {result};` since the return type is `std::optional<T>` and `result` is of type `T`. I can see your point about braces looking a bit odd here: feel free to change :)

Some context behind the "braced return" style: it can help avoid nasty implicit narrowing conversions like in the following example.

```
$ cat without-list-initialization.cpp
#include <cstdint>

struct F {
  F(uint8_t) {}
};

F f() { return 1024; }
$ clang++ -c without-list-initialization.cpp
# No errors
$ echo $?
0
```

```
$ cat with-list-initialization.cpp
#include <cstdint>

struct F {
  F(uint8_t) {}
};

F f() { return {1024}; }
$ clang++ -c with-list-initialization.cpp
with-list-initialization.cpp:7:18: error: constant expression evaluates to 1024 which cannot be narrowed to type 'uint8_t' (aka 'unsigned char') [-Wc++11-narrowing]
$ echo $?
1
```

See item (8) and "narrowing conversions" in https://en.cppreference.com/w/cpp/language/list_initialization :)

Reply:
FWIW also: https://clang.llvm.org/extra/clang-tidy/checks/modernize-return-braced-init-list.html

Reply:
> FWIW also: https://clang.llvm.org/extra/clang-tidy/checks/modernize-return-braced-init-list.html


`return {baz};` fails when `explicit Foo(Baz)`